__tests__/endpoints.test.js

```diff
@@ -25,13 +27,37 @@ afterAll(() => {
                 expect(body.topics.length).toBe(3)
             })
         })
-        test("* returns 404 not found for invalid url", () => {
-            return request(app)
-            .get('/api/topicssss')
-            .expect(404)
-            .then(({body}) => {
-                expect(body.msg).toBe('path not found')
+    })
+})
+
+describe("test for invalid paths", () => {
+    test("* returns 404 not found for invalid url", () => {
+        return request(app)
+        .get('/api/topicssss')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('path not found')
+        })
+    })
+})
+
+describe("GET '/api' ", () => {
+    test("GET /api returns information to the user about all available endpoints", () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then(({body}) => {
+            const endpointsKeys = Object.keys(endpointsObject)
+            
+            expect(body.endpoints["GET /api"]).toMatchObject({
+                description: expect.any(String)
+            })
+            expect(body.endpoints["GET /api/topics"]).toMatchObject({
+                description: expect.any(String),
+                queries:  expect.any(Array),
+                exampleResponse: expect.any(Object)
             })
+            expect(Object.keys(body.endpoints)).toEqual(endpointsKeys)
```

Comment:
Here, you test the first two key value pairs, and then that all the keys are equal. This is functional, but not very complete, as you will keep updating the endpoints.json so a lot of the values will not be part of your equality tests. Can you simplify this,  to check that the whole body.endpoints object`Equal`s the endpoints object you've imported into your test suite? 👀